chain/near/src/chain.rs

```diff
@@ -348,3 +361,146 @@ impl FirehoseMapperTrait<Chain> for FirehoseMapper {
+            SuccessValueExecutionStatus,
+        },
+        Chain,
+    };
+
+    use super::TriggersAdapter;
+
+    #[tokio::test]
+    async fn test_trigger_filter_empty() {
+        let account1: String = "account1".into();
+
+        let adapter = TriggersAdapter {};
+
+        let logger = Logger::root(slog::Discard, o!());
+        let block1 = random_block(1, &account1);
+
+        let filter = TriggerFilter::default();
+
+        let block_with_triggers: BlockWithTriggers<Chain> = adapter
+            .triggers_in_block(&logger, block1, &filter)
+            .await
+            .expect("failed to execute triggers_in_block");
+        assert_eq!(block_with_triggers.trigger_count(), 0);
+    }
+
+    #[tokio::test]
+    async fn test_trigger_filter_every_block() {
+        let account1: String = "account1".into();
+
+        let adapter = TriggersAdapter {};
+
+        let logger = Logger::root(slog::Discard, o!());
+        let block1 = random_block(1, &account1);
+
+        let filter = TriggerFilter {
+            block_filter: crate::adapter::NearBlockFilter {
+                trigger_every_block: true,
+            },
+            ..Default::default()
+        };
+
+        let block_with_triggers: BlockWithTriggers<Chain> = adapter
+            .triggers_in_block(&logger, block1, &filter)
+            .await
+            .expect("failed to execute triggers_in_block");
+        assert_eq!(block_with_triggers.trigger_count(), 1);
+
+        let height: Vec<u64> = heights_from_triggers(&block_with_triggers);
+        assert_eq!(height, vec![1]);
+    }
+
+    #[tokio::test]
+    async fn test_trigger_filter_every_receipt() {
+        let account1: String = "account1".into();
+
+        let adapter = TriggersAdapter {};
+
+        let logger = Logger::root(slog::Discard, o!());
+        let block1 = random_block(1, &account1);
+
+        let filter = TriggerFilter {
+            receipt_filter: NearReceiptFilter {
+                accounts: HashSet::from_iter(vec![account1]),
+            },
+            ..Default::default()
+        };
+
+        let block_with_triggers: BlockWithTriggers<Chain> = adapter
+            .triggers_in_block(&logger, block1, &filter)
+            .await
+            .expect("failed to execute triggers_in_block");
+        assert_eq!(block_with_triggers.trigger_count(), 1);
+
+        let height: Vec<u64> = heights_from_triggers(&block_with_triggers);
+        assert_eq!(height.len(), 0);
+    }
+
+    fn heights_from_triggers(block: &BlockWithTriggers<Chain>) -> Vec<u64> {
+        block
+            .trigger_data
+            .clone()
+            .into_iter()
+            .filter_map(|x| match x {
+                crate::trigger::NearTrigger::Block(b) => b.header.clone().map(|x| x.height),
+                _ => None,
+            })
+            .collect()
+    }
+
+    fn random_block(height: u64, receiver_id: &String) -> codec::Block {
+        codec::Block {
```

Comment:
Beautiful!

Reply:
I expected it to have randomness given the name. I much prefer `handcrafted_artisan_rust_sourced_block`.

Reply:
🤣  I'll rename